scripts/backfill_source_tags_for_unregistered_contributors.py

```diff
@@ -0,0 +1,181 @@
+from datetime import timedelta
+import logging
+from django.apps import apps
+from website.app import init_app
+from scripts import utils as script_utils
+import datetime
+import progressbar
+from django.core.paginator import Paginator
+
+import sys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def backfill_source_tags_for_osf4m_unregistered_contributors(dry_run):
+    """ Backfill osf4m source tags to all osf4m unregistered contributors
+    """
+    Tag = apps.get_model('osf', 'Tag')
+    OSFUser = apps.get_model('osf', 'OSFUser')
+    NodeLog = apps.get_model('osf', 'NodeLog')
+    meeting_source_tag = Tag.all_tags.get(name='source:campaign|osf4m', system=True)
+    meeting_nodes = meeting_source_tag.abstractnode_tagged.all()
+    logging.info('Number of meeting nodes found: ' + str(len(meeting_nodes)))
+    all_meeting_node_logs = NodeLog.objects.filter(action='contributor_added', node__in=meeting_nodes).only('created', 'params')
+    ThroughModel = OSFUser.tags.through
+    set_of_user_ids = set()
+    pbar = progressbar.ProgressBar(maxval=len(all_meeting_node_logs) or 1).start()
+    pbarcounter = 0
+    for entry in all_meeting_node_logs:
+        pbarcounter += 1
+        entry_created_date = entry.created
+        for contributor_id in entry.params['contributors']:
+            contributor_added = OSFUser.objects.get(guids___id=contributor_id)
+            if contributor_added.is_invited and contributor_added.date_confirmed and contributor_added.date_confirmed > entry_created_date:
+                set_of_user_ids.add(contributor_added.pk)
+                # If the user is merged to another user, add the same tag to that user as well
+                if contributor_added.merged_by is not None:
+                    set_of_user_ids.add(contributor_added.merged_by.pk)
+        pbar.update(pbarcounter)
+    logging.info('Number of meeting nodes unreg contrib found: ' + str(len(set_of_user_ids)))
+    if not dry_run:
+        list_of_user_ids_already_with_osf4m_source_tag = OSFUser.objects.filter(tags__id=meeting_source_tag.id).values_list('pk', flat=True)
+        for id in list_of_user_ids_already_with_osf4m_source_tag:
```

Comment:
Removes the primary keys of users already with this tag because of `bulk_create`.